app/src/main/java/com/epfl/beatlink/ui/components/library/TrackVoteCard.kt

```diff
@@ -1,62 +1,74 @@
 import androidx.compose.material3.CardDefaults
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
 import com.epfl.beatlink.R
-import com.epfl.beatlink.model.spotify.objects.SpotifyTrack
+import com.epfl.beatlink.model.library.PlaylistTrack
 import com.epfl.beatlink.ui.components.VoteButton
-import com.epfl.beatlink.ui.theme.NegativeGradientBrush
 import com.epfl.beatlink.ui.theme.PositiveGradientBrush
 import com.epfl.beatlink.ui.theme.TypographySongs
 
 @Composable
-fun TrackVoteCard(spotifyTrack: SpotifyTrack) {
+fun TrackVoteCard(
+    playlistTrack: PlaylistTrack,
+    onVoteChanged: (String, Boolean) -> Unit,
+    userId: String
+) {
   Card(
       modifier = Modifier.fillMaxSize().testTag("trackVoteCard"),
       colors = CardDefaults.cardColors(containerColor = Color.Transparent)) {
         Row(modifier = Modifier.fillMaxSize(), verticalAlignment = Alignment.CenterVertically) {
-          Image(
-              painter = painterResource(id = R.drawable.cover_test1), // TODO
-              contentDescription = "Playlist cover",
-              modifier = Modifier.size(55.dp).padding(horizontal = 6.dp))
-          // Playlist details
+          // Album cover
+          Box(
+              modifier =
+                  Modifier.size(55.dp)
+                      .padding(horizontal = 5.dp, vertical = 5.dp)
+                      .clip(RoundedCornerShape(4.dp))) {
+                AsyncImage(
+                    model = playlistTrack.track.cover,
+                    contentDescription = "Cover for ${playlistTrack.track.name}",
+                    modifier = Modifier.fillMaxSize(),
+                    contentScale = ContentScale.Crop)
+              }
+          // Track details
           Column(modifier = Modifier.weight(1f)) {
             Text(
-                text = spotifyTrack.name,
+                text = playlistTrack.track.name,
                 style = TypographySongs.titleLarge,
                 color = MaterialTheme.colorScheme.primary)
             Text(
-                text = spotifyTrack.artist,
+                text = playlistTrack.track.artist,
                 style = TypographySongs.titleMedium,
             )
           }
 
-          VoteButton(
-              NegativeGradientBrush,
-              MaterialTheme.colorScheme.primary,
-              painterResource(R.drawable.boring_face))
-          Spacer(Modifier.width(10.dp))
+          // Vote button
           VoteButton(
               PositiveGradientBrush,
               MaterialTheme.colorScheme.secondary,
```

Comment:
You can directly put the gradient and the color variables in the VoteButton composable since we only use one button to vote.